torch/lib/c10d/Utils.hpp

```diff
@@ -464,6 +474,25 @@ using SizeType = uint64_t;
 // `success_cond` is an expression used to check if an error has happend. So for
 // `fork()`, we can use `SYSCHECK(pid = fork(), pid != -1)`. The function output
 // is stored in variable `__output` and may be used in `success_cond`.
+#ifdef _WIN32
+#define SYSCHECK(expr, success_cond)                                         \
+  while (true) {                                                             \
+    auto __output = (expr);                                                  \
+    auto errno_local = WSAGetLastError();                                    \
+    (void)__output;                                                          \
+    if (!(success_cond)) {                                                   \
+      if (errno == EINTR) {                                                  \
+        continue;                                                            \
+      } else if (errno_local == EAGAIN || errno_local == EWOULDBLOCK) {      \
```

Comment:
Here `EWOULDBLOCK` should be `WSAEWOULDBLOCK`. It's not the same value. According to the [doc](https://docs.microsoft.com/en-us/windows/win32/winsock/windows-sockets-error-codes-2), `WSAEWOULDBLOCK` is 10035, while `EWOULDBLOCK` is `140` in `errno.h`.

Same for `EINTR` and `WSAEINTR`.

And `EAGAIN` should probably be `WSAETIMEDOUT`.

Reply:
WSAEINTR and EINTR has same value.
Resolved